contracts/core/exchange-wrappers/TakerWalletWrapper.sol

```diff
@@ -77,39 +78,31 @@ contract TakerWalletWrapper is
                 takerTokenAmount := mload(add(_orderData, add(scannedBytes, 32)))
             }
 
-            executeTransfer(
+            // Transfer from taker's wallet to this wrapper
+            ITransferProxy(transferProxy).transfer(
+                takerToken,
+                takerTokenAmount,
                 _taker,
-                _tradeOriginator,
+                address(this)
+            );
+
+            // Ensure allowance of transfer from this wrapper to TransferProxy
+            ERC20Wrapper.ensureAllowance(
                 takerToken,
+                address(this),
+                transferProxy,
                 takerTokenAmount
             );
 
+            // Record taker token and amount to return values
+            uint256 orderCount = scannedBytes >> 6;
```

Comment:
How does this work? Does it increment one at a time?

Reply:
Yes, but this is particular to our workflow we scan 64 bytes at at time, so you can think of it as counting from 0 starting on the 7th bit.